src/Test.UnitTests.Sarif/TestData/RoundTripping/Inputs/InvocationOverridesRuleDefaultConfigurationOfError.sarif

```diff
@@ -0,0 +1,61 @@
+{
+  "$schema": "https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.5.json",
+  "version": "2.1.0",
+  "runs": [
+    {
+      "tool": {
+        "driver": {
+          "name": "FailureLevelRoundtrippingTester",
+          "rules": [
+            {
+              "id": "ExplicitDefaultLevel",
+              "defaultConfiguration": { "level": "error" }
+            }
+          ]
+        }
+      },
+      "invocations": [
+        {
+          "ruleConfigurationOverrides": [
+            {
+              "descriptor": {
+                "id": "ExplicitDefaultLevel"
+              },
+              "configuration": {
+                "level": "note"
+              }
+            }
+          ],
+          "executionSuccessful": true
+        }
+      ],
+      "results": [
+        {
+          "ruleId": "ExplicitDefaultLevel/ShouldBeNone",
+          "level": "none",
+          "message": { "text": "Level should be 'none' as it is directly expressed on the result. This is a rule trace message." }
+        },
+        {
+          "ruleId": "ExplicitDefaultLevel/ShouldBeNone",
+          "kind": "open",
+          "message": { "text": "Level should be 'none' because kind is not 'fail' and level is missing (so 'none' is the default)." }
+        },
+        {
+          "ruleId": "ExplicitDefaultLevel/ShouldBeNote",
+          "message": { "text": "Level should be 'note' as level is missing and 'note' is the invocation override configuration default." }
+        },
```

Comment:
I think we can also add a explicit kind fail test,
we know it should have the same result but if we add we know it is tested.

Reply:
See other comment. These files *must* be rendered precisely as we expect them to be when our serializer writes them to disk (according to the spec, eliding default values). That invariant is required in order to author simple, general purpose validation logic.

1) We read the log file
2) We validate that all result level/kind values are correct
3) We save the log file again to disk
4) We read that saved copy and validate nothing has changed.